File: src/main/java/net/lucgameshd/easymysql/serialization/LocalDateDeserializer.java
package net.lucgameshd.easymysql.serialization;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;

import java.io.IOException;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class LocalDateDeserializer extends JsonDeserializer<LocalDate> {
    @Override
    public LocalDate deserialize( JsonParser jsonParser, DeserializationContext deserializationContext ) throws IOException {
        return LocalDate.parse( jsonParser.getValueAsString(), DateTimeFormatter.ofPattern( "yyyy-MM-dd" ) );
    }
}

File: src/main/kotlin/net/lucgameshd/easymysql/serialization/LocalDateDeserializer.kt
package net.lucgameshd.easymysql.serialization

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer
import java.time.LocalDate
import java.time.format.DateTimeFormatter

class LocalDateDeserializer : JsonDeserializer<LocalDate>() {
    override fun deserialize(p: JsonParser, ctxt: DeserializationContext): LocalDate {
        return LocalDate.parse(p.valueAsString, DateTimeFormatter.ofPattern("yyyy-MM-dd"))
    }
}
